app/models/concerns/indexable.rb

```diff
@@ -33,13 +33,15 @@ def set_published_at
       unless published_at
         if defined?(metadata) && metadata && metadata['date']
           begin
-            update_column(:published_at, metadata['date'])
-          rescue ActiveRecord::StatementInvalid
-            update_column(:published_at, created_at)
+            self.published_at = Date.parse(metadata['date'])
+          rescue StandardError
+            self.published_at = created_at
```

Comment:
Hmmm, the only way I can think of right now is to set the date to the January 1st if only the year is available. Do you think that would be good enough for the search results? Probably better than using the `created_at` anyway.